.github/workflows/continuous_integration.yml

```diff
@@ -22,3 +22,6 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - uses: psf/black@23.11.0
+      - uses: civerachb-cpr/find-trailing-whitespace@master
```

Comment:
[discussion] I don't agree with applying this requirement across the entire code base. We don't have formatting requirements on contrib scripts. This is to lower the barrier to entry for contributors. See https://github.com/Allen-Synthesis/EuroPi/blob/main/contributing.md#code-style-requirements

Reply:
I suppose that leads to the obvious question of whether or not EuroPi has matured to the point where a slightly higher bar is warranted? I'd be in favour of expanding the `black` linting to the whole `software` directory and enforcing a somewhat more uniform coding standard.

I assume the decision to have a low barrier for contributors was done to encourage the community to write scripts to contribute when EuroPi first launched?  Has the module & community matured to the point where being a little pickier is good for the sake of better consistency?

As for this specific change, most IDEs I've used (e.g. VS Code, Atom, Geany, Vim) will automatically remove trailing whitespace on save. It's a common enough feature that I was actually surprised to see there were so many instances of it throughout the code. As far as adding a barrier for new contributors, I think this is a very low one.

Reply:
I think the fact the community has matured is a reason to *maintain* the lower barrier for entry - more people are joining who will have less programming ability, as in the first place before the firmware was developed as fully, you had to have a certain level of skill to begin writing.

I've wondered in the past if black formatting could be implemented as a git hook when PRs are submitted/commits are made on a PR, that way script writers/editors don't even need to be aware it happens, and when they pull main it'll be formatted correctly.

Even though I agree that for you and I it's a small barrier, for someone who has just begun programming and has written something simple, I can see them having to first get their head around how EuroPiScript works as a class, and *then* how to format their PR descriptively enough to be accepted, and *then* be told the whitespace at the end of their lines isn't acceptable.

I think this is part of a larger discussion, but I would like this project to favour beginners and hobbyists rather than professionalism. There's a balance to be struck, but I would rather PRs sometimes re-add or remove trailing whitespace and it's a tiny bit ugly for reviewers to look at in the raw diff, than anyone see a huge list of PR requirements (even if individually they're very simple)